src/helpers/clientChronicles.js

```diff
@@ -0,0 +1,193 @@
+import axios from 'axios';
+import {ulid} from 'ulid'
+
+const CHRONICLES_BACKEND = process.env.REACT_APP_CHRONICLES_BACKEND;
+const chroniclesUrl = path => `${CHRONICLES_BACKEND}${path}`;
+
+//An array of DOM events that should be interpreted as user activity.
+const ACTIVITY_EVENTS = ['mousedown', 'mousemove', 'keydown', 'scroll', 'touchstart'];
+
+const FLOWS = [
+  {start: 'page-enter',               end: 'page-leave',                 subFlows: []},
+  {start: 'unit-page-enter',          end: 'unit-page-leave',            subFlows: ['player-play']},
+  {start: 'collection-page-enter',    end: 'collection-page-leave',      subFlows: ['collection-unit-selected']},
+  {start: 'collection-unit-selected', end: 'collection-unit-unselected', subFlows: ['player-play']},
+  {start: 'player-play',              end: 'player-stop',                subFlows: []},
+];
+
+const FLOWS_BY_END = new Map(FLOWS.map(flow => [flow.end, flow]));
+const SUBFLOWS = new Map(Object.entries(FLOWS.reduce((acc, flow) => {
+  flow.subFlows.forEach(subflow => {
+    if (!(subflow in acc)) {
+      acc[subflow] = [];
+    }
+    if (!acc[subflow].includes(flow.start)) {
+      acc[subflow].push(flow.start);
+    }
+  });
+  return acc;
+}, {})));
+
+const MAX_INACTIVITY_MS = 60 * 1000; // Minute in milliseconds.
+
+export default class ClientChronicles {
+  constructor(history) {
+    if (!CHRONICLES_BACKEND) {
+      return;
+    }
+    const localStorage = window.localStorage;
+    if (localStorage.getItem('user_id') === null) {
+      localStorage.setItem('user_id', `local:${ulid()}`);
+    }
+    this.userId = localStorage.getItem('user_id');
+
+    this.namespace = 'archive';
+
+    this.initSession(/* reinit= */ false);
+
+    // Setup the setInterval method to run every second.
+    setInterval(() => {
+      // Check for inactivity, only if was active since last inactive timestamp.
+      if (this.isUserActive() && Date.now() - this.lastActivityTimestampMs > MAX_INACTIVITY_MS) {
+        // if the user has been inactive or idle for longer then the seconds specified in MAX_INACTIVITY_MS.
+        console.log(`User has been inactive for more than ${MAX_INACTIVITY_MS} seconds.`);
+        this.append('user-inactive', {activities: Array.from(this.sessionActivities)});
+      }
+    }, 1000);
```

Comment:
Are you sure it will make owners of low-end phones happy?

Reply:
This is a background 1 second delay computation without any http calls so it is very light...
Once in 2 minutes will send http request.

Reply:
FYI, I'm planning in the future to "bulk" the appends and send them once every ~10 seconds, to reduce http overhead and server side handling.

But I don't want to do premature optimizations as this might not be required. WDYT?

Reply:
Append does send http request right away.